lockfile/src/javascript.rs

```diff
@@ -71,6 +71,11 @@ impl Parse for PackageLock {
                     continue;
                 }
 
+                // Ignore extraneous dependencies.
+                if keys.get("extraneous").is_some() {
+                    continue;
```

Comment:
Skipping over it looks good, but what about the second part of the issue's expected behavior:

> we should skip over it and print a warning to the user letting them know they should run npm prune to clean up these packages.

Reply:
We cannot print warnings directly from the parsers, so that gets a bit tricky.

Silently ignoring them matches the behavior of other cases, such as bundled or vendored dependencies. With no install step, they pose minimal security risk.